languages/python/django-oso/django_oso/partial.py

```diff
@@ -91,17 +91,23 @@ def and_expr(expr: Expression, model: Model, **kwargs):
     return q
 
 
-def compare_expr(expr: Expression, _model: Model, path=(), **kwargs):
+def compare_expr(expr: Expression, model: Model, path=(), **kwargs):
     q = Q()
     (left, right) = expr.args
     left_path = dot_path(left)
     if left_path:
         return COMPARISONS[expr.operator](q, "__".join(path + left_path), right)
     else:
-        if isinstance(right, Model):
+        if isinstance(right, model):
             right = right.pk
         else:
+            # QUESTION: Should this just return an empty set? Similar to how invalid isa
+            # returns empty set.
```

Comment:
Feels different to me -- the semantics of returning an empty set would be "this is a valid query that returns no results," which I don't think is an accurate representation of what's happening.

Reply:
Not sure... see my example above.  Unifying something with a type that doesn't match would end up in this branch, which is very similar to checking isa with a type that doesn't match.

Reply:
In light of the above realization, I agree that this should probably return `Q(pk__in=[])`, which is Django's version of the "always false" SQL clause. (See usage in `isa_expr()`, above.)